apps/dashboard/src/components/contract-components/fetchPublishedContractsFromDeploy.ts

```diff
@@ -17,11 +24,36 @@ export async function fetchPublishedContractsFromDeploy(options: {
     throw new Error("No IPFS URI found in bytecode");
   }
 
-  const publishURIs = await getPublishedUriFromCompilerUri({
+  let publishURIs = await getPublishedUriFromCompilerUri({
     contract: getContractPublisher(client),
     compilerMetadataUri: contractUri,
   });
 
+  // Try fetching using contract name from compiler metadata for zksolc variants
+  // TODO: ContractPublisher should handle multiple metadata uri for a published version
+  if (publishURIs.length === 0 && (await isZkSyncChain(contract.chain))) {
+    const deployMetadata = await download({
+      uri: contractUri,
+      client,
+    })
+      // biome-ignore lint/suspicious/noExplicitAny:
+      .then((res) => res.json() as any)
```

Comment:
You can type this, just do as <the minimal object you need> doesn't have to be the whole thing.

Also don't we already have a type for compiler meta ?

Reply:
We have types for parsed metadata, but those don't match the exact metadata formate generated by compiler, esp zksolc.